cmd/apprepository-controller/server/controller.go

```diff
@@ -417,10 +418,41 @@ func ownerReferencesForAppRepo(apprepo *apprepov1alpha1.AppRepository, childName
 	return nil
 }
 
+// intervalToCron transforms string durations like "1m" or "1h" to cron expressions
+// Even if valid time units are "ns", "us", "ms", "s", "m", "h",
+// the result will get rounded up to minutes.
+func intervalToCron(duration string) (string, error) {
+	if duration == "" {
+		return "", fmt.Errorf("duration cannot be empty")
+	} else {
+		if d, err := time.ParseDuration(duration); err != nil {
+			return "", err
+		} else {
+			cronMins := math.Ceil(d.Minutes()) // round up to nearest minutes
+			// https://kubernetes.io/docs/concepts/workloads/controllers/cron-jobs/#cron-schedule-syntax
+			return fmt.Sprintf("*/%v * * * *", cronMins), nil // every cronMins minutes
+		}
+	}
+}
+
 // newCronJob creates a new CronJob for a AppRepository resource. It also sets
 // the appropriate OwnerReferences on the resource so handleObject can discover
 // the AppRepository resource that 'owns' it.
 func newCronJob(apprepo *apprepov1alpha1.AppRepository, config Config) *batchv1.CronJob {
+	var err error
+
+	cronTime := config.Crontab
+
+	// If the apprepo has its own interval,
+	// use that instead of the default global crontab.
+	if apprepo.Spec.Interval != "" {
+		cronTime, err = intervalToCron(apprepo.Spec.Interval)
+	}
+	// If the interval is invalid, use the default global crontab
+	if err != nil {
```

Comment:
Hmm - I know we don't yet have any way to report this back to the user (other than failing the operation, which isn't great either), so I reckon it'd be useful to at least log this error so that we can help debug if there's an issue?